lib/wallet-e2e/src/Cardano/Wallet/Spec/Effect/Assert.hs

```diff
@@ -2,38 +2,32 @@
 
 module Cardano.Wallet.Spec.Effect.Assert where
 
-import qualified Effectful.Error.Dynamic as Effect
+import qualified Effectful.Error.Static as Effect
 
 import Cardano.Wallet.Spec.Effect.Trace
     ( FxTrace, trace )
+import Control.Exception
+    ( AssertionFailed (..) )
 import Effectful
     ( (:>), Eff, Effect )
 import Effectful.Dispatch.Dynamic
     ( interpret )
-import Effectful.Error.Dynamic
+import Effectful.Error.Static
     ( throwError )
 import Effectful.TH
     ( makeEffect )
 import Prelude hiding
     ( trace )
-import Text.Show
-    ( show )
-
-newtype Error = Error Text
-    deriving newtype (Eq)
-
-instance Show Error where
-    show (Error msg) = "Assertion failed: " <> toString msg
 
 data FxAssert :: Effect where
     Assert :: Text -> Bool -> FxAssert m ()
 
 $(makeEffect ''FxAssert)
 
 runAssertError
```

Comment:
how you choose between interpret and reinterpret ?

Reply:
reinterpret introduces a handler-local effect, while interpret works with the effects already declared.

One example where reinterpret is useful is introducing a handler-local state.

In this case we interpret Assert effect in terms of the Error effect (global), which in turn is handled elsewhere.